src/Umbraco.Web.UI.Client/src/common/services/mediahelper.service.js

```diff
@@ -5,10 +5,37 @@
 **/
 function mediaHelper(umbRequestHelper, $log) {
 
-    //container of fileresolvers
+    // container of fileresolvers, file resolvers are only useful for full media object representations (i.e. content based), they are
+    // not used for entity (light weight) based media objects (i.e. resolved from the entity resource).
     var _mediaFileResolvers = {};
 
     return {
+
+        /**
+         * Used when retreiving a media item from the entity resource to format it's data so that it contains the properties that
+         * other services and views are expecting. 
+         * @param {any} entity
+         */
+        formatMediaEntityData: function (mediaEntity) {
+            if (mediaEntity && mediaEntity.metaData) {
+
+                if (mediaEntity.metaData.IsFolder !== null && mediaEntity.metaData.IsFolder !== undefined) {
+                    mediaEntity.isFolder = mediaEntity.metaData.IsFolder;
+                }
+
+                if (mediaEntity.metaData.MediaPath) {
+                    mediaEntity.thumbnail = this.resolveFileFromEntity(mediaEntity, true);
+                    mediaEntity.image = this.resolveFileFromEntity(mediaEntity, false);
+                    if (mediaEntity.isFolder === undefined) {
+                        mediaEntity.isFolder = mediaEntity.image ? false : true;
+                    }
+                }
+                else if (mediaEntity.isFolder === undefined) {
+                    mediaEntity.isFolder = mediaEntity.metaData.ContentTypeAlias && mediaEntity.metaData.ContentTypeAlias.toLowerCase() === "folder";
```

Comment:
This may be too restrictive as it prevents additional, bespoke folder types being catered for. It could at least be based on these types having aliases beginning with 'folder', i.e., 
`mediaEntity.isFolder = mediaEntity.metaData.ContentTypeAlias && mediaEntity.metaData.ContentTypeAlias.toLowerCase().startsWith("folder");`